app/src/main/kotlin/com/wire/android/ui/calling/CallActivity.kt

```diff
@@ -0,0 +1,142 @@
+
+    @Inject
+    lateinit var proximitySensorManager: ProximitySensorManager
+
+    private val qualifiedIdMapper = QualifiedIdMapperImpl(null)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        callNotificationManager.hideAllNotifications()
+
+        setUpCallingFlags()
+
+        appLogger.i("$TAG Initializing proximity sensor..")
+        proximitySensorManager.initialize()
+
+        WindowCompat.setDecorFitsSystemWindows(window, false)
+
+        val conversationId = intent.extras?.getString(EXTRA_CONVERSATION_ID)
+        val screenType = intent.extras?.getString(EXTRA_SCREEN_TYPE)
+
+        setContent {
+            val snackbarHostState = remember { SnackbarHostState() }
+            CompositionLocalProvider(
+                LocalSnackbarHostState provides snackbarHostState,
+                LocalActivity provides this
+            ) {
+                WireTheme {
+                    conversationId?.let {
+                        screenType?.let { screenType ->
+                            val startDestination = CallScreenType.valueOf(screenType)
+                            CallScreen(
+                                conversationId = qualifiedIdMapper.fromStringToQualifiedID(it),
+                                startDestination = startDestination
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        proximitySensorManager.registerListener()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        proximitySensorManager.unRegisterListener()
+    }
+
+    companion object {
+        private const val TAG = "CallActivity"
+        const val EXTRA_CONVERSATION_ID = "conversation_id"
+        const val EXTRA_USER_ID = "user_id"
+        const val EXTRA_SCREEN_TYPE = "screen_type"
+    }
+}
+
+fun CallActivity.setUpCallingFlags() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
+        setShowWhenLocked(true)
+        setTurnScreenOn(true)
+    } else {
+        window.addFlags(
+            WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
+                    or WindowManager.LayoutParams.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON,
+        )
+    }
+}
+
+fun getOngoingCallIntent(
+    activity: Activity,
+    conversationId: String
+) = Intent(activity, CallActivity::class.java).apply {
+    putExtra(CallActivity.EXTRA_CONVERSATION_ID, conversationId)
+    putExtra(CallActivity.EXTRA_SCREEN_TYPE, CallScreenType.Ongoing.name)
+}
+
+fun getInitiatingCallIntent(
+    activity: Activity,
+    conversationId: String
+) = Intent(activity, CallActivity::class.java).apply {
+    putExtra(CallActivity.EXTRA_CONVERSATION_ID, conversationId)
+    putExtra(CallActivity.EXTRA_SCREEN_TYPE, CallScreenType.Initiating.name)
+}
+
+fun CallActivity.openAppLockActivity() {
```

Comment:
move it to Activity, what the point of having it as extended?

Reply:
To be specific to only CallActivity